src/change_stream/mod.rs

```diff
@@ -0,0 +1,177 @@
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target.
+///
+/// `ChangeStream`'s are "resumable", meaning that they can be restarted at a given place in the
+/// stream of events. This is done automatically when the `ChangeStream` encounters certain
+/// ["resumable"](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#resumable-error) errors, such as transient network failures. It can also be done manually by passing
+/// a [`ResumeToken`](document/struct.ResumeToken.html) retrieved from a past event
+/// into either the
+/// [`resume_after`](option/struct.ChangeStreamOptions.html#structfield.resume_after)
+/// or [`start_after`](option/struct.ChangeStreamOptions.html#structfield.start_after)
+/// (4.2+) options used to create the `ChangeStream`. Issuing a raw change stream aggregation is
+/// discouraged unless users wish to explicitly opt out of resumability.
+///
+/// A `ChangeStream` can be iterated to return batches of instances of any type that implements the
+/// Deserialize trait. By default, it returns batches of
+/// [`ChangeStreamEventDocument`](document/struct.ChangeStreamEventDocument.html).
+/// These documents correspond to changes in the associated collection, database or deployment.
+///
+/// A `ChangeStream` can be iterated like any other `Iterator`:
+///
+/// ```rust
+/// # #[cfg(not(feature = "sync"))]
+/// # use futures::stream::StreamExt;
+/// # use mongodb::{Client, error::Result, bson::doc,
+/// # change_stream::document::ChangeStreamEventDocument};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// task::spawn(async move {
+///     coll.insert_one(doc! { "x": 1 }, None).await;
+/// });
+/// let mut change_stream = coll.watch(None, None).await?;
+/// while let Some(change) = change_stream.next().await {
+///     let document: ChangeStreamEventDocument = change?;
+///     println!("operation performed: {:?}, document: {:?}", document.operation_type, document.full_document);
+///     // operation performed: Insert, document: Some(Document({"x": Int32(1)}))
+/// }
+/// #
+/// # Ok(())
+/// # }
+/// ```
+///
+/// A `ChangeStream` allows any Deserialize type for `ChangeStreamEventDocument` to be specified:
+/// ```rust
+/// # #[cfg(not(feature = "sync"))]
+/// # use bson::Document;
+/// # use futures::stream::StreamExt;
+/// # use serde::Deserialize;
+/// # use mongodb::{Client, error::Result, bson::doc, change_stream::document::ChangeStreamEventDocument};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// # #[derive(Deserialize, Debug)]
```

Comment:
we want these `derive`s to show up in the documentation, so the leading `#` can be removed.

Reply:
fixed